libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsRouteEx.kt

```diff
@@ -45,9 +47,23 @@ internal fun List<com.mapbox.navigator.Waypoint>.mapToSdk(): List<Waypoint> =
             internalType = nativeWaypoint.type.mapToSdk(),
             name = nativeWaypoint.name,
             target = nativeWaypoint.target,
+            metadata = nativeWaypoint.metadata?.parseMetadata()
         )
     }
 
+private fun String.parseMetadata(): Map<String, JsonElement>? {
```

Comment:
I there a chance that this function will be called from the main thread?

Reply:
Yes. It's a small json though.
Right now nativeWaypoints is being lazily initialized. So it can be invoked at any time from any thread.
But maybe it's better to move this parsing to route creation (it has a lot of parsing anyway). And these waypoints we'll be accessed anyway.